src/System.Net.Sockets.Legacy/tests/FunctionalTests/SendReceive.cs

```diff
@@ -0,0 +1,555 @@
+                    receiverAck.Reset();
+                    senderAck.Set();
+
+                    Assert.Equal(DatagramSize, sent);
+                    sentChecksums[sentDatagrams] = Fletcher32.Checksum(sendBuffer, 0, sent);
+
+                    sentDatagrams++;
+                    if (sentDatagrams == DatagramsToSend)
+                    {
+                        right.Dispose();
+                        return;
+                    }
+                }
+                else
+                {
+                    sentDatagrams = 0;
+                }
+
+                random.NextBytes(sendBuffer);
+                sendBuffer[0] = (byte)sentDatagrams;
+                right.SendToAPM(sendBuffer, 0, sendBuffer.Length, SocketFlags.None, leftEndpoint, sendHandler);
+            };
+            sendHandler(0);
+
+            receiverFinished.Task.Wait();
+            for (int i = 0; i < DatagramsToSend; i++)
+            {
+                Assert.Equal(sentChecksums[i], receivedChecksums[i]);
+            }
+        }
+
+        private static void SendRecv_Stream_TCP(IPAddress listenAt, bool useMultipleBuffers)
+        {
+            const int BytesToSend = 123456;
+
+            var server = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+            server.BindToAnonymousPort(listenAt);
+
+            server.Listen(128);
+
+            int bytesReceived = 0;
+            var receivedChecksum = new Fletcher32();
+            var serverThread = new Thread(() =>
+            {
+                using (server)
+                {
+                    Socket remote = server.Accept();
+                    Assert.NotNull(remote);
+
+                    using (remote)
+                    {
+                        if (!useMultipleBuffers)
+                        {
+                            var recvBuffer = new byte[256];
+                            for (;;)
+                            {
+                                int received = remote.Receive(recvBuffer, 0, recvBuffer.Length, SocketFlags.None);
+                                if (received == 0)
+                                {
+                                    break;
+                                }
+                                bytesReceived += received;
+                                receivedChecksum.Add(recvBuffer, 0, received);
+                            }
+                        }
+                        else
+                        {
+                            var recvBuffers = new List<ArraySegment<byte>> {
+                                new ArraySegment<byte>(new byte[123]),
+                                new ArraySegment<byte>(new byte[256], 2, 100),
+                                new ArraySegment<byte>(new byte[1], 0, 0),
+                                new ArraySegment<byte>(new byte[64], 9, 33)
+                            };
+                            for (;;)
+                            {
```

Comment:
Formatting issues are fixed. `for (;;)` is what I'm used to (it's slightly terser than `while (true)`).